platine-management-service/src/main/java/fr/insee/survey/datacollectionmanagement/questioning/service/impl/QuestioningServiceImpl.java

```diff
@@ -114,26 +121,30 @@ public Set<Questioning> findBySurveyUnitIdSu(String idSu) {
     /**
      * Generates an access URL based on the provided parameters.
      *
-     * @param baseUrl      The base URL for the access.
-     * @param typeUrl      The type of URL (V1 or V2).
      * @param role         The user role (REVIEWER or INTERVIEWER).
      * @param questioning  The questioning object.
-     * @param surveyUnitId The survey unit ID.
+     * @param part          Part of questioning
      * @return The generated access URL.
      */
-    public String getAccessUrl(String baseUrl, String typeUrl, String role, Questioning questioning, String surveyUnitId, String sourceId) {
+    public String getAccessUrl(String role, Questioning questioning, Partitioning part) {
+        String typeUrl = parametersService.findSuitableParameterValue(part, ParameterEnum.URL_TYPE);
+        String surveyUnitId = questioning.getSurveyUnit().getIdSu();
         // Set default values if baseUrl or typeUrl is empty
-        baseUrl = StringUtils.defaultIfEmpty(baseUrl, questioningUrl);
         typeUrl = StringUtils.defaultIfEmpty(typeUrl, V3.name());
 
         if (typeUrl.equalsIgnoreCase(V1.name())) {

```

Comment:
suggestion: could use a switch pattern on typeUrl => will handle all enum cases and could be returned directly